Solutions/Zscaler Private Access (ZPA)/Hunting Queries/ZscalerAbnormalTotalBytesSize.yaml

```diff
@@ -0,0 +1,31 @@
+id: 181dc982-4631-11ec-81d3-0242ac130003
+name: Zscaler - Abnormal total bytes size
+description: |
+  'Query shows abnormal total bytes size.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: ZscalerPrivateAccess
+    dataTypes:
+      - ZPAEvent
+tactics:
+  - Exfiltration
+  - Collection
+relevantTechniques:
+  - T1530
+  - T1537
+query: |
+  let Average_Body_Bytes = ZPAEvent
+  | where TimeGenerated > ago(14d)
+  | summarize Avg_Size = avg(tolong(ZENTotalBytesRxConnector))
+  | extend K = 1;
+  ZPAEvent
+  | where TimeGenerated > ago(7d)
+  | extend K = 1
+  | join kind=inner Average_Body_Bytes on K

```

Comment:
There is no need for a (very heavy) join here. Just convert the Average_Body_Bytes to a scalar: let let Average_Body_Bytes = toscalar (....); and use it as a number in the main query.

Reply:
fixed